File: ios/Models/DeviceProfile.swift
import Foundation

struct DeviceProfile: Codable, Identifiable {
    enum Tier: String, Codable {
        case a = "A"
        case b = "B"
        case c = "C"
    }

    enum RuntimePreference: String, Codable {
        case coreml
        case tfliteIos
    }

    let id: String
    let osVersion: String
    let chipset: String
    let thermalThresholds: [String: Double]
    let batteryCapacityMah: Int
    let tier: Tier
    let estimatedFps: Double
    let defaultRuntime: RuntimePreference
    let lastEvaluatedAtMillis: Int64
}

File: android/app/src/main/java/com/golfiq/hud/model/DeviceProfile.kt
package com.golfiq.hud.model

import com.golfiq.hud.inference.RuntimeMode

data class DeviceProfile(
    val id: String,
    val osVersion: String,
    val chipset: String,
    val thermalThresholds: Map<String, Double>,
    val batteryCapacityMah: Int,
    val tier: Tier,
    val estimatedFps: Double,
    val defaultRuntime: RuntimeMode,
    val lastEvaluatedAtMillis: Long,
) {
    enum class Tier { A, B, C }
}
